StandaloneMmPkg/Drivers/StandaloneMmIplPei/MmFoundationHob.c

```diff
@@ -843,23 +916,33 @@ CreateMmFoundationHobList (
     UsedSize += HobLength;
   }
 
+  //
+  // Build resource HOB for unblocked region
+  //
   HobLength = GetRemainingHobSize (*FoundationHobSize, UsedSize);
-  if (PcdGetBool (PcdCpuSmmRestrictedMemoryAccess)) {
+  MmIplBuildResourceHobForUnblockedRegion (FoundationHobList + UsedSize, &HobLength);
+  UsedSize += HobLength;
+
+  if (!PcdGetBool (PcdCpuSmmRestrictedMemoryAccess)) {
+    UnblockRegion = NULL;
     //
-    // Only unblocked memory regions are accessible
+    // Collect unblocked memory regions
     //
-    MmIplBuildResourceHobForUnblockedRegion (FoundationHobList + UsedSize, &HobLength);
-  } else {
+    UnblockRegion = CollectUnblockedMemoryRegions (&UnblockRegionCount);
     //
     // All system memory (DRAM) is accessible.
     // When SMM Profile is enabled:
     //   * Access to regions reported from MmPlatformHobProducerLib do not require logging.
     //   * Access to other system memory requires logging.
     //
-    MmIplBuildResourceHobForAllSystemMemory (FoundationHobList + UsedSize, &HobLength, PlatformHobList, PlatformHobSize, Block, MmProfileDataHob);
-  }
+    HobLength = GetRemainingHobSize (*FoundationHobSize, UsedSize);
+    MmIplBuildResourceHobForAllSystemMemory (FoundationHobList + UsedSize, &HobLength, PlatformHobList, PlatformHobSize, Block, MmProfileDataHob, UnblockRegion, UnblockRegionCount);

```

Comment:
MmIplBuildResourceHobForAllSystemMemory (
  FoundationHobList + UsedSize,
  &HobLength,
  PlatformHobList,
  PlatformHobSize,
  Block,
  MmProfileDataHob,
  UnblockRegion,
  UnblockRegionCount
  );

Reply:
Updated in new commit